### PR TITLE
[CIRCSTORE-170] Add Item Limit to Loan Policy

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
**Purpose**
Add a new field to `Loan Policy`.

**Link**
[CIRCSTORE-170](https://issues.folio.org/browse/CIRCSTORE-170)

### CODE REVIEW

File: src/test/java/org/folio/rest/api/LoanPoliciesApiTest.java [hunk 1]
```diff
@@ -314,6 +314,93 @@ public void cannotCreateALoanPolicyWithAdditionalPropertiesInLoanPolicy()
...
+
+    loanPolicyRequest.getJsonObject("loansPolicy").put("itemLimit", 1000);
+
+    client.post(loanPolicyStorageUrl(),
+      loanPolicyRequest, StorageTestSuite.TENANT_ID,
+      ResponseHandler.json(createCompleted));
+
+    JsonResponse response = createCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(String.format("Failed to create loan policy: %s", response.getBody()),
```
Comment (bohdan-suprun) [inline src/test/java/org/folio/rest/api/LoanPoliciesApiTest.java]: Could you please add an assertion to check that the property is really got persisted (i.e. it returned in post response, or read the policy by ID and verify that the property is present)?

File: src/test/java/org/folio/rest/api/LoanPoliciesApiTest.java [hunk 2]
```diff
@@ -314,6 +314,93 @@ public void cannotCreateALoanPolicyWithAdditionalPropertiesInLoanPolicy()
...
+    loanPolicyRequest.getJsonObject("loansPolicy").put("itemLimit", 0);
+
+    client.post(loanPolicyStorageUrl(),
+      loanPolicyRequest, StorageTestSuite.TENANT_ID,
+      ResponseHandler.json(createCompleted));
+
+    JsonResponse response = createCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(String.format("Should fail to create loan policy: %s", response.getBody()),
+      response.getStatusCode(), is(HttpStatus.HTTP_INTERNAL_SERVER_ERROR.toInt()));
```
Comment (bohdan-suprun) [inline src/test/java/org/folio/rest/api/LoanPoliciesApiTest.java]: I think it make sense to assert error message as well, in order to make sure that the error is caused by validation rather then another error.

Hmm... another important note, why do we not expect validation error (422) in this case, but server error instead?

Could you please check if it is something related to https://issues.folio.org/browse/RMB-503 that @mreno-EBSCO observed? He has posted a temp fix for this issue.

File: src/test/java/org/folio/rest/api/LoanPoliciesApiTest.java (hunk 1 above)
Comment (OleksandrVidinieiev) [reply src/test/java/org/folio/rest/api/LoanPoliciesApiTest.java]: done

File: src/test/java/org/folio/rest/api/LoanPoliciesApiTest.java (hunk 2 above)
Comment (OleksandrVidinieiev) [reply src/test/java/org/folio/rest/api/LoanPoliciesApiTest.java]: Adding this missing dependency did help, thanks!

File: ramls/loan-policy.json [hunk 3]
```diff
@@ -44,6 +44,12 @@
         "fixedDueDateScheduleId": {
           "type": "string",
           "description": "Fixed due date schedule (due date limit)"
+        },
+        "itemLimit": {
```
Comment (marcjohnson-kint) [inline ramls/loan-policy.json]: What is the item limit? Is it the number of items a patron can have checked out at the same time?

Comment (OleksandrVidinieiev) [reply ramls/loan-policy.json]: Hi @marcjohnson-kint, please find a detailed explanation of what an `Item limit` is in the description of [CIRC-558](https://issues.folio.org/browse/CIRC-558).

**Pull request merged**